### PR TITLE
Update everything to Node 8

### LINKED ISSUE #1030: Update to Node 8
We're using Node 6 in our docker images (and in cloud.gov), yet we're also using package-lock.json files, which aren't part of the bundled npm until Node 8. Let's upgrade everything to the latest LTS and remove the shrinkwrap files.

### PR DESCRIPTION
**Pull request by cmc333333**
The Node 8 image uses a new version of `npm`, which removes the `npm-shrinkwrap.json` file in favor of `package-lock.json` (which now also includes dev dependencies).

Unfortunately, our binary libraries will likely fail out due to being built for Node 6. To resolve, it's easiest to:
```sh
rm -rf */node_modules
# these commands can be ran in parallel
bin/api-npm install
bin/styleguide-npm install
bin/ui-npm install
```
Note that, particularly the ui step, may take a while (~20m) to run.

This resolves #1030

### CODE REVIEW

File: api/.docker/deps_ok_then [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 if [ ! -x node_modules/.bin/deps-ok ]; then
   echo "Installing deps-ok"
-  npm install --quiet deps-ok
+  npm install --save-dev --quiet deps-ok
```
Comment (toolness) [inline api/.docker/deps_ok_then]: Sorry, why are we doing `--save-dev` now?

**Comment by toolness** [comment]
Oh, also, I just noticed that we're now using CircleCI's fancy caching mechanism as of #999, which I _think_ should mean that CircleCI logs shouldn't be very noisy if we remove the `--quiet` from all our `npm install` invocations (except for when we clear the CircleCI cache).

As I mentioned in #588, I think that removing the `--quiet` would really help responsiveness for developers who are onboarding (or who have just blown away their `node_modules`). Do you think we should do that here or a separate PR?

File: api/.docker/deps_ok_then (hunk 1 above)
Comment (cmc333333) [reply api/.docker/deps_ok_then]: The `deps-ok` lib (which confirms top-level modules are installed) is only used during development. Previously, we were running `npm install` _without_ `--save`, which would add the lib to our environment but not affect our package/shrinkwrap files. The new version of npm automatically updates these files, however, so we need to place it differently.

**Comment by cmc333333** [comment]
Good call, happy to remove the `--quiet`s now

**Pull request merged**